src/lib/utils/api/nav.ts

```diff
@@ -1,11 +1,23 @@
-import { IndexApi } from './index-api'
+import { getIndexTokenData } from '@indexcoop/tokenlists'
+
+import {
+  IndexDataMetric,
+  IndexDataProvider,
+} from '@/lib/utils/api/index-data-provider'
 
 export class NavProvider {
   async getNavPrice(tokenSymbol: string, chainId = 1): Promise<number> {
-    const indexApi = new IndexApi()
-    const symbol = tokenSymbol.replace(/-/g, '').toLowerCase()
-    const path = `/${chainId}/${symbol}/nav`
-    const res = await indexApi.get(path)
-    return res.nav
+    const token = getIndexTokenData(tokenSymbol, chainId)
+    if (!token) throw new Error(`Index token not found: ${tokenSymbol}`)
+
+    const indexDataProvider = new IndexDataProvider()
+    const res = await indexDataProvider.getTokenMetrics({
+      tokenAddress: token.address,
+      chainId,
+      metrics: [IndexDataMetric.Nav],
+    })
+
+    // TODO: Determine if this should be 0 or throw
```

Comment:
@janndriessen thoughts on this line/behavior? e.g. what to do if we encounter an exception when we're fetching the `nav`. The coingecko API seems to default to 0 in that case, but i'm not sure if it wouldn't behave as well here